### PR TITLE
test: improve DTMF test to wait for canInsertDTMF

### PR DESCRIPTION
**Pull request by fippo**
using the pattern from https://github.com/w3c/web-platform-tests/blob/master/webrtc/RTCDTMFSender-helper.js

### CODE REVIEW

**Comment by fippo** [comment]
@alvestrand can you take a look? Also happy to contribute the small style improvements back to wpt ;-)

**Comment by fippo** [comment]
and ofc the flakyness in chrome stable triggers :joy:

**Comment by fippo** [comment]
https://github.com/w3c/webrtc-pc/issues/1619#issuecomment-366963505 suggests this should not be necessary but practically speaking it makes sense.

File: test/e2e/dtmf.js [hunk 1]
```diff
@@ -104,7 +104,22 @@ describe('dtmf', () => {
...
+            'canInsertDTMF' in window.RTCDTMFSender.prototype)) {
+          return;
+        }
+        return new Promise((resolve) => {
+          setTimeout(function canInsert() {
+            const sender = pc1.getSenders().find(s => s.track.kind === 'audio');
+            if (sender.dtmf.canInsertDTMF) {
+              return resolve();
+            }
+            setTimeout(canInsert, 10);
```
Comment (jan-ivar) [inline test/e2e/dtmf.js:117]: This is a terrible API IMHO. `canInsertDTMF` should be for determining whether DTMF is possible in the current configuration, not for figuring out ***when*** to start to send DTMF.

I think we need to fix the spec to specify when this boolean can be expected to be true, e.g. in the `"connected"` event (from https://github.com/w3c/webrtc-pc/issues/1619#issuecomment-366963505).

Comment (fippo) [reply test/e2e/dtmf.js:117]: it is but this test has been flaky forever. I still think there is a chrome bug here but we'll see

Comment (alvestrand) [reply test/e2e/dtmf.js:117]: The code turned out to eat DTMF sent before connection was in ready state, which was an even more terrible API.
The thing that SHOULD work is to not look until connection is established - but even waiting for state changes had issues.

**Review by alvestrand — APPROVED** [review]

**Pull request merged**